### ISSUE TITLE
Add support for ARM servers

### ISSUE THREAD

**Issue by NoaHimesaka1873** [Jigsaw-Code/outline-server#1009]
Tried to run this on ARM server, it hanged on
```
 > Waiting for Outline server to be healthy ....
```
so I checked docker log and...
```
standard_init_linux.go:228: exec user process caused: exec format error
```
this happened. Seems like binaries don't support aarch64. Looks like adding aarch64-compatible binaries will solve this problem.

**Comment by Prajwal-Koirala** [comment]
I support this completely.

**Comment by neneeen** [comment]
https://github.com/Jigsaw-Code/outline-server/issues/244#issuecomment-646507320 works for me on Oracle's ARM VPS

**Comment by kugaevsky** [comment]
You could try my build with arm64 support https://github.com/kugaevsky/outline-server-arm64

**Comment by jason05031** [comment]
root@instance-20220330-2007:~# sudo SB_IMAGE="kugaevsky/outline-server-arm64" bash -c "$(wget -qO- https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh)"
Unsupported machine type: aarch64. Please run this script on a x86_64 machine

Sorry! Something went wrong. If you can't figure this out, please copy and paste all this output into the Outline Manager screen, and send it to us, to see if we can help you.
Full log: /tmp/outline_logVG4

**Comment by tnduc** [comment]
same issue

**Comment by pvp-by** [comment]
+up
```sudo SB_IMAGE="kugaevsky/outline-server-arm64" bash -c "$(wget -qO- https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh)"
Unsupported machine type: aarch64. Please run this script on a x86_64 machine```
not working 06.sept.22

**Comment by kugaevsky** [comment]
https://github.com/kugaevsky/outline-server-arm64/issues/3#issuecomment-1069800724

**Comment by seia-soto** [comment]
Here's simpler solution.

`curl -sL "https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh" | sed -e '/Unsupported machine type/{N;d;}' | bash`

- https://github.com/seia-soto/outline-server-multiarch/pkgs/container/shadowbox#unsupported-machine-type-machine_type-please-run-this-script-on-a-x86_64-machine

**Comment by c0state** [comment]
I've been watching this and other ARM support issues/PRs for years(!) and took another look just now. I was able to get this to work, somewhat satisfactorily, using https://github.com/tonistiigi/binfmt, on my Raspberry Pi 4 8GB running Ubuntu Server

In short, 
1. Install the emulator via `docker run --privileged --rm tonistiigi/binfmt --install amd64` (for amd64 emulation)
	- You might need to run this via systemd (or whatever your system requires) as it doesn't persist on reboot
	- I followed https://askubuntu.com/a/919059/41028
2. Retrieve the install script via `wget https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh` 
	- Remove the lines at https://github.com/Jigsaw-Code/outline-server/blob/6ea0c1d084ab064f945078c0038fa56acf61c4a5/src/server_manager/install_scripts/install_server.sh#L417-L422
	  (the command at https://github.com/Jigsaw-Code/outline-server/issues/1009#issuecomment-1284197880 should *currently* work, but I did it manually)
3. Finally, run the install script: `sudo bash -c "./install_server.sh --hostname <ddns_hostname> --api-port abc --keys-port xyz"` (I pass hostname since I have a Dynamic DNS name for my server and also specific ports, but you can leave these all out)

Unsure how much slower this is. Will need to test this a bit

**Comment by pps83** [comment]
So, it's still not supported? Really, no arm version?

**Comment by Infinity6542** [comment]
No, not officially at least. I managed to get the latest version of Outline (1.12.3) running on an OCI instance using https://github.com/pmh-only/shadowbox-arm64-image.

Just had to run
```
export SB_IMAGE="ghcr.io/pmh-only/shadowbox:1.12.3"

curl -sL "https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh" | sed '/local MACHINE_TYPE/,/fi/{d}' | bash
```

This is an unofficial image tho and there are problems with it, the biggest being seemingly no way to access or modify keys. I can create them, but I can't actually get the key nor change the default port/data limit. I was able to do it using https://github.com/AmRo045/OutlineAdmin, however, so it may just be an issue with Outline Manager on my end? I have another server running 1.7.2 (https://github.com/seia-soto/outline-server-multiarch) and that seems to work just fine...

**Comment by pps83** [comment]
> No, not officially at least. 

FYI, this one works for me on arm: https://github.com/kugaevsky/outline-server-arm64

**Comment by Infinity6542** [comment]
I've made a fork of the previous image I mentioned, available at https://github.com/Infinity6542/shadowbox-arm64-image (I've included installation instructions too). It should give you the latest version of Outline

**Comment by levanid** [comment]
> I've made a fork of the previous image I mentioned, available at https://github.com/Infinity6542/shadowbox-arm64-image (I've included installation instructions too). It should give you the latest version of Outline

Thank you, at last I managed to update to the latest version on aarch64 and it seems to work as intended 🙏🏻
